lambdas/utils/logging_formatter.py

```diff
@@ -6,19 +6,28 @@
 
 class LoggingFormatter(logging.Formatter):
     def format(self, record: logging.LogRecord) -> str:
-
         auth = "No Auth"
-        if request_context.authorization is not None:
+        request_id = "No Request"
+        patient_nhs_no = "No NHS number"
+        app_interaction = "Missing app interaction"
+        if getattr(request_context, "authorization", None) is not None:
             auth = request_context.authorization
+        if getattr(request_context, "request_id", None) is not None:
+            request_id = request_context.request_id
+        if getattr(request_context, "patient_nhs_no", None) is not None:
+            patient_nhs_no = request_context.patient_nhs_no
+        if getattr(request_context, "app_interaction", None) is not None:
+            app_interaction = request_context.app_interaction
 
-        s = super().format(record)
         d = {
-            "correlation_id": request_context.request_id,
-            "auth": auth,
-            "Message": s
+            "Message": record.getMessage(),
+            "App Interaction": app_interaction,
+            "Correlation Id": request_id,
+            "Authorisation": auth,
+            "Patient NHS number": patient_nhs_no,
         }
 
-        if record.__dict__.get("custom_args", {}) is not None: 
+        if record.__dict__.get("custom_args", {}) is not None:
             d.update(record.__dict__.get("custom_args", {}))
 
         return json.dumps(d)
```

Comment:
I think here we can just put the default value inside the 3rd argument of getattr, instead of doing default=None and then compare it with None
Also, maybe better to give a nicer name to the dictionary `d`

For example, it could be like:
```
    def format(self, record: logging.LogRecord) -> str:
        auth = getattr(request_context, "authorization", "No Auth")
        request_id = getattr(request_context, "request_id", "No Request")
        patient_nhs_no = getattr(request_context, "patient_nhs_no", "No NHS number")
        app_interaction = getattr(request_context, "app_interaction", "Missing app interaction")

        log_content = {
            "Message": record.getMessage(),
            "App Interaction": app_interaction,
            "Correlation Id": request_id,
            "Authorisation": auth,
            "Patient NHS number": patient_nhs_no,
        }

        if record.__dict__.get("custom_args", {}) is not None:
            log_content.update(record.__dict__.get("custom_args", {}))

        return json.dumps(log_content)
```